easybuild/tools/systemtools.py

```diff
@@ -1118,6 +1118,9 @@ def get_linked_libs_raw(path):
     or None for other types of files.
     """
 
+    if os.path.islink(path):
+        _log.debug(f"{path} is a symbolic link, so skipping check for linked libs")
+        return None
```

Comment:
I think we should:
* resolve the symlink here so the `file` call below makes sense;
* do the skipping of symlinks where we're calling `get_linked_libs_raw` instead (since otherwise we're checking duplicate files);

If I call `get_linked_libs_raw` on any path that corresponds to a binary (either directly or via a symlink), I expect to get a lit of linked libraries, not `None` when it's a symlink.

Reply:
I didn't want to change existing behavior. Resolving the symlink here instead of at the call site might discover issues previously not seen and fail builds.

But yes skipping symlinks at the callsite makes sense. As written in the description we would then need(?) to remove the check for symlinks pointing outside the install directory as if we skip symlinks that doesn't make sense anymore.

Can do of course if the behavior change is acceptable.

Reply:
We don't need to resolve the link here at all?

We just check whether the path is a symlink at the call site, rather than here.

If someone (in a different context) calls `get_linked_libs_raw` with a (single) path that corresponds to a symlink, they should should get a proper answer, not `None` if it happens to be a symlink (the function isn't called `get_linked_libs_raw_only_for_real_binaries_not_symlinks` ;) )

Reply:
Hold on, I overlooked something...

Because of the `file` command that we run on the specified path, we're already getting `None` as result from calling `get_linked_libs_raw` on a symlinked path, so all you're doing here is adding a short-circuit.

Sorry for the confusion, ignore the above. :)